presto-main/src/main/java/com/facebook/presto/operator/ExchangeClientFactory.java

```diff
@@ -66,7 +67,11 @@ public ExchangeClientFactory(
         this.concurrentRequestMultiplier = concurrentRequestMultiplier;
         this.minErrorDuration = requireNonNull(minErrorDuration, "minErrorDuration is null");
         this.httpClient = requireNonNull(httpClient, "httpClient is null");
-        this.maxResponseSize = requireNonNull(maxResponseSize, "maxResponseSize is null");
+
+        requireNonNull(maxResponseSize, "maxResponseSize is null");
+        long maxResponseSizeBytes = Math.min(httpClient.getMaxContentLength(), maxResponseSize.toBytes()) / 2;
```

Comment:
What about this change? https://github.com/miniway/presto/commit/d93e4cc1e2b2d5492eb78187cb17ca0f82904fd0 

Is it somewhat overkill?